10.0 wip format

PeterAlabaster opened the pull request.

Commit:
[META] Port report menu additions to 10

Also I have restructured the module to have a normal structure.
The report definitions for the 3 reports which were in version 8 are missing
Also, there are reports which will be added to this repo which I have not yet
created skeletons for.
Just milestoning progress at this point

Commit:
[ADD] New external layout and UK headers done

May be some slight tweaks to this, but for the most part i'll consider the
headers done. The templating and framework for footers is also here now.

Commit:
[ADD] Footer skeleton, and t and c fields for footer

Commit:
[ADD] Footer definitions for all skeletions

Commit:
[META] Adding todos due to misinformation and changes required

Commit:
[FIX] Remove doc specific footer items.

This is due to a re-work in the spec, which involves moving T&Cs to body.
Other parts of footer have also been removed, thus rendering it nice and generic

Commit:
[ADD] Purchase report body, and header/footer tweaks

Note that large spacing between header and body is due to paperformat
header spacing integer. To make this look 'right' I set it to 25 on the
paperformat European A4

Commit:
[ADD] Change to description column, table width

Commit:
[ADD] New paperformat. Minor tweak to po lines

Commit:
[ADD] RFQ template, generic templates, minor styling changes

Commit:
[META] Refactor to remove unnecessary field

Commit:
[ADD] Spacing between orderlines table rows

Commit:
[ADD] Quotation and SO reports

Commit:
[FIX] Make total line headers bold

Commit:
[META] Remove now unnecessary comments of old code

Commit:
[META] Tidy unnecessary open tag

Commit:
[ADD] Invoice template. Tweak VAT / desc on SO / Inv

Commit:
[ADD] Proforma report body

Commit:
[REM] More deprecated comments

Commit:
[ADD] Credit note body

Commit:
[FIX] use of is_integer does not work on integers.

Commit:
[ADD] Delivery note report

Commit:
[FIX] Space main blocks evenly where there are only 2 blocks

i.e on invoice reports and delivery note

Commit:
[FIX] Move t&c blocks to new row

Commit:
[REM] account statement report and update manifest

Commit:
[META] Make some more generic templates for partner_id and headers

Also add in a header for all invoice reports

Commit:
[ADD] Payment to information block to invoice / proforma

Commit:
[FIX] Base delivery date on date of transfer, not scheduled date

Commit:
[META] Reformat docstrings

Commit:
[META] Make all of footer text line italic

Commit:
[META] Remove unnecessary deprecated comment

Commit:
[META] Convert to tabs

james-opusvl commented on UK_Reports/__manifest__.py:
```diff
@@ -0,0 +1,61 @@
+
+{
+	'name': 'UK customised reports',
+	'version': '10.0.0.1',
```
```suggestion
	'version': '10.0.1.0.0',
```

Commit:
Update UK_Reports/__manifest__.py version number

Co-Authored-By: PeterAlabaster <peter.alabaster@opusvl.com>

james-opusvl commented on UK_Reports/models/product_product.py:
```diff
@@ -20,19 +20,17 @@
+	_inherit = "product.product"
 
+	def _vendor_specific_code(self, vendor):
+		my_vendors = self.seller_ids.filtered(lambda seller: seller.name == vendor)
```
Docstring saying vendor is a res.partner.id

Commit:
[META] Update README

Commit:
[META] Add docstring for _vendor_product_code

Commit:
[FIX] Migrate int check to helper function

Commit:
[FIX] Helper function doesn't need `self`

james-opusvl commented on UK_Reports/models/res_company.py:
```diff
@@ -0,0 +1,81 @@
+class ResCompany(models.Model):
+	_inherit = "res.company"
+
+	accounts_dept_phone = fields.Char(
```
Adding the following as it prompted a discussion in the office. I feel the 'Odoo' way of doing this would be to create related fields to company>partner>contact(type=invoicing). That would mean that new contact types need to be created as part of this module.

No change required now, but to be discussed further in the future.

Commit:
[META] Intentation failure

Commit:
[FIX] USe super calls on print button

Also, use correct report IDs instead of the old format

Commit:
[META] Use super call on account invoice print button too

james-opusvl commented on UK_Reports/reports/account_invoice_report.xml:
```diff
@@ -0,0 +1,196 @@
+			<t t-call="UK_Reports.template_document_title"/>
+			<div class="row">
+				<!-- Space the address to the right slightly -->
+				<div class="col-xs-1"/>
```
```suggestion
```

james-opusvl commented on UK_Reports/reports/account_invoice_report.xml:
```diff
@@ -0,0 +1,196 @@
+			<div class="row">
+				<!-- Space the address to the right slightly -->
+				<div class="col-xs-1"/>
+				<div class="col-xs-5">
```
```suggestion
				<div class="col-xs-5 col-offset-1">
```

PeterAlabaster replied:
Applied suggestion (after trialling the 'proper' way to do it) on all instances where offset was done manually with an empty div. see e2a44f1

Commit:
[META] Use col-offset instead of empty div

Commit:
[META] Move paperformat to load last. Remove deprecated report from pf

james-opusvl commented on UK_Reports/reports/external_layouts.xml:
```diff
@@ -0,0 +1,111 @@
+				<span class="source-mini-icon-text mail" t-esc="email"/>
+				<br/>
+			</div>
+			<div class="col-xs-5 source-address-column">
```
columns add up to 13?

PeterAlabaster replied:
Applied fix here: a9347f7

james-opusvl commented on UK_Reports/static/src/css/UK_Reports-body.css:
```diff
@@ -0,0 +1,21 @@
+table.orderlines-totals {
```
```suggestion
table.orderlines-totals, table.orderlines {
```

PeterAlabaster replied:
Applied suggestion. See df59dff

james-opusvl commented on UK_Reports/static/src/css/UK_Reports-body.css:
```diff
@@ -0,0 +1,21 @@
+table.orderlines-totals {
+	width: 100%;
+}
+table.orderlines {
```
```suggestion

```

james-opusvl commented on UK_Reports/static/src/css/UK_Reports-body.css:
```diff
@@ -0,0 +1,21 @@
+	width: 100%;
+}
+table.orderlines {
+	width: 100%;
```
```suggestion

```

james-opusvl commented on UK_Reports/static/src/css/UK_Reports-body.css:
```diff
@@ -0,0 +1,21 @@
+}
+table.orderlines {
+	width: 100%;
+}
```
```suggestion

```

james-opusvl reviewed (commented):
See inline comments

Nick-OpusVL commented:
On 17/04/2019 15:49, James Curtis wrote:
> In UK_Reports/models/product_product.py 
> <https://github.com/OpusVL/Odoo-UK-Format-Reports/pull/1#discussion_r276281395>:
> 
>>  
> +	def _vendor_specific_code(self, vendor):
> +		my_vendors = self.seller_ids.filtered(lambda seller: seller.name == vendor)
> 
> Docstring saying vendor is a res.partner.id

A res.partner actually looking at the way the vendor variable is used.

Either way I wouldn't encourage passing integer ids anywhere unless it 
is absolutely necessary.

Commit:
[META] Two css elements had same attributes. Merged them

Commit:
[META] Make row 12 columns, remove left shift css for class

CSS was saying to shift it to the left, which is why we used 13 columns (to
shift it back to the right). Silly mistake
